scipy/io/matlab/mio4.py

```diff
@@ -287,12 +287,12 @@ def shape_from_header(self, hdr):
             # Read only the row and column counts
             self.mat_stream.seek(dt.itemsize * (dims[0] - 1), 1)
             rows = np.ndarray(shape=(1,), dtype=dt,
-                              buffer=self.mat_stream.read(dt.itemsize))
+                              buffer=self.mat_stream.read(dt.itemsize))[0]
             self.mat_stream.seek(dt.itemsize * (dims[0] - 1), 1)
             cols = np.ndarray(shape=(1,), dtype=dt,
-                              buffer=self.mat_stream.read(dt.itemsize))
+                              buffer=self.mat_stream.read(dt.itemsize))[0]
 
-            shape = (int(rows), int(cols))
```

Comment:
Instead of changing this line, why not change `(1,)` to `()` above?

Reply:
Ah right; I had thought that `hdr.dtype` was int-like, but it's not. Fixed.